bindings/python/requirements.txt

```
@@ -2,6 +2,5 @@
 
 numpy>=1.17.4
 
-open-space-toolkit-core~=2.1
-open-space-toolkit-io~=2.0
-open-space-toolkit-mathematics~=2.0
+open-space-toolkit-io~=2.0.0
```

```suggestion
open-space-toolkit-core~=2.1.0
open-space-toolkit-io~=2.0.0
```

I think we should leave the explicit core dependency, as we have it in the cpp

---

Yep!

---

Wait actually, couldn't we remove the dependency in cpp as well? @kyle-cochran That way we don't have this double dependency?

---

No, unfortunately. Cpp doesn't have a smart dependency resolver like Python does with `pip`. Unless you bake in the required dependency code to your library (this is what compiling _statically_ provides) you have to:
- download
- version control
- link
every single requirement manually. Tools like CMake make this a little easier, but it's still a hassle.

For example, we could statically compile Core. That way, when IO is compiled, all of the Core code is included in the binary, then Physics would have access to it through the IO binary (although linking to the correct function would be a hassle).

Usually you just have shared objects `.so` files. That way both the Physics _and_ IO binaries just use the same one.

---

Ah ok gotcha, makes sense